pilot/pkg/networking/core/v1alpha3/listener_test.go

```diff
@@ -40,47 +42,66 @@ func TestOutboundListenerConflict_HTTP(t *testing.T) {
 	// The oldest service port is TCP.  We should encounter conflicts when attempting to add the HTTP ports. Purposely
 	// storing the services out of time order to test that it's being sorted properly.
 	testOutboundListenerConflict(t,
-		buildService(model.ProtocolHTTP, tnow.Add(1*time.Second)),
-		buildService(model.ProtocolTCP, tnow),
-		buildService(model.ProtocolHTTP, tnow.Add(2*time.Second)))
+		buildService("test1,com", model.ProtocolHTTP, tnow.Add(1*time.Second)),
+		buildService("test2,com", model.ProtocolTCP, tnow),
+		buildService("test3,com", model.ProtocolHTTP, tnow.Add(2*time.Second)))
 }
 
 func TestOutboundListenerConflict_TCP(t *testing.T) {
 	// The oldest service port is HTTP.  We should encounter conflicts when attempting to add the TCP ports. Purposely
 	// storing the services out of time order to test that it's being sorted properly.
 	testOutboundListenerConflict(t,
-		buildService(model.ProtocolTCP, tnow.Add(1*time.Second)),
-		buildService(model.ProtocolHTTP, tnow),
-		buildService(model.ProtocolTCP, tnow.Add(2*time.Second)))
+		buildService("test1.com", model.ProtocolTCP, tnow.Add(1*time.Second)),
+		buildService("test2.com", model.ProtocolHTTP, tnow),
+		buildService("test3.com", model.ProtocolTCP, tnow.Add(2*time.Second)))
 }
 
 func TestOutboundListenerConflict_Unordered(t *testing.T) {
 	// Ensure that the order is preserved when all the times match. The first service in the list wins.
 	testOutboundListenerConflict(t,
-		buildService(model.ProtocolHTTP, tzero),
-		buildService(model.ProtocolTCP, tzero),
-		buildService(model.ProtocolTCP, tzero))
+		buildService("test1.com", model.ProtocolHTTP, tzero),
+		buildService("test2.com", model.ProtocolTCP, tzero),
+		buildService("test3.com", model.ProtocolTCP, tzero))
 }
 
-func testOutboundListenerConflict(t *testing.T, services ...*model.Service) {
-	t.Helper()
-
+func TestMergeOutboundTCPFilterChains(t *testing.T) {
```

Comment:
I would park this test until the code to emit distinct filter chains based on destination CIDRs goes in.
At that time, you should be checking for number of filter chains in the listner and possibly even looking at the filter chain match conditions to see if the IPs are there.

Reply:
I would just expect the author of that PR to update this test. Let's just get these tests in so we have coverage.